### PR TITLE
fix: remove hard coded contig names from gcnv wrappers (#489 )

### PR DESCRIPTION
**Pull request by Nicolai-vKuegelgen**

### CODE REVIEW

File: snappy_wrappers/wrappers/gcnv/contig_ploidy/wrapper.py [hunk 1]
```diff
@@ -43,14 +43,24 @@
...
+# Get contig name style
+grep "SN:chr" {snakemake.input.interval_list} > /dev/null && true
+exit_status=$?
+if [[ $exit_status == 0 ]]; then
+    STYLE="chr"
+else
+    STYLE=""
+fi
+
+
```
Comment (tedil) [inline snappy_wrappers/wrappers/gcnv/contig_ploidy/wrapper.py:55]: I don't know what the input interval_list looks like; I assume/hope it does not mix "chr" and "" prefixes ;)
Personally, I would try to make sure that the "SN:chr" is a bit more specific (i.e. start of line, followed by a number or X/Y and/or has trailing newline/tab/space/whatever).

File: snappy_wrappers/wrappers/gcnv/post_germline_calls/wrapper.py [hunk 2]
```diff
@@ -25,14 +25,25 @@
     fi
 done
 
+# Get contig name style
+# Note: the sample_index in '{snakemake.input.ploidy}' and '{snakemake.input.calls[0]}' are NOT guaranteed to be the same
+# -> Sample_0 should always exist and the contig names (which we care for) are the same
+tail -n +2 $(dirname {snakemake.input.ploidy})/ploidy-calls/SAMPLE_0/contig_ploidy.tsv | grep "chr" > /dev/null && true
```
Comment (tedil) [inline snappy_wrappers/wrappers/gcnv/post_germline_calls/wrapper.py:31]: See the other comment about being more specific with the grep pattern, might be fine, as long as the input really has no stray "chr" that is not a contig prefix for whatever reason.

**Review by tedil — CHANGES_REQUESTED** [review]
As I don't know what the input looks like, the grep patterns might be too unspecific.
Other than that, looks good to me.
(Though I would like to remove code duplication and ideally find a better way of handling this. A dedicated 'give me the prefix'/'give me the sex chromosome names' rule whose output serves as additional input/param for other rules.)

File: snappy_wrappers/wrappers/gcnv/contig_ploidy/wrapper.py (hunk 1 above)
Comment (Nicolai-vKuegelgen) [reply snappy_wrappers/wrappers/gcnv/contig_ploidy/wrapper.py:55]: I've specified the regex, and I think if the file would have mixed prefixes a lot of other things would also break, so I'm not too concerned about check for that here.

File: snappy_wrappers/wrappers/gcnv/post_germline_calls/wrapper.py (hunk 2 above)
Comment (Nicolai-vKuegelgen) [reply snappy_wrappers/wrappers/gcnv/post_germline_calls/wrapper.py:31]: Also made this regex more specific

**Review by tedil — APPROVED** [review]

**Pull request merged**